optics/src/Optics.hs

```
@@ -224,52 +199,356 @@ import Data.Either.Optics                    as P
+-- established many of the foundations on which the @optics@ package builds (not
+-- least in quite a bit of code having been directly ported).  It defines optics
+-- based on the /van Laarhoven/ representation, where each optic kind is
+-- introduced as a /transparent/ type synonym for a complex polymorphic type,
+-- for example:
+--
+-- @
+-- type Lens s t a b = forall f. 'Functor' f => (a -> f b) -> s -> f t
+-- @
+--
+-- In contrast, @optics@ tries to preserve an abstraction boundary between the
+-- interface of optics and their implementation.  Optic kinds are expressed
+-- directly in the types, as 'Optic' is an /opaque/ newtype:
+--
+-- @
+-- type 'Lens' s t a b = 'Optic' 'A_Lens' 'NoIx' s t a b
+-- @
+--
+-- The choice of representation of 'Optic' is then an implementation detail, not
+-- essential for understanding the library.  (In fact, @optics@ uses the
+-- /profunctor/ representation rather than the /van Laarhoven/ representation;
+-- this affects the optic kinds and operations that can be conveniently
+-- supported, but not the essence of the design.)
 --
--- In general, this leads to better results from type inference (the optic kind
--- is preserved in the inferred type):
+-- Our design choice to use /opaque/ rather than /transparent/ abstractions
+-- leads to various consequences, both positive and negative, which are explored
+-- in the following subsections.
+--
+-- == Advantages of the opaque design
+--
+-- Since the interface is deliberately chosen rather than to some extent
+-- determined by the implementation, we are free to choose a more restricted
+-- interface where doing so leads to conceptual simplicity.  For example, in
+-- @lens@, the 'view' function can be used with a 'Fold' provided the result
+-- type has a 'Monoid' instance, and the multiple targets of the 'Fold' will be
+-- combined monoidally.  This behaviour can be confusing, so in @optics@ a
+-- 'Getter' cannot be silently used as a 'Fold', and we prefer to have 'view'
```

I think you meant the other way around, i.e. "a Fold cannot be silently used as a Getter".

---

Well spotted! Fixed.